x-pack/plugins/lens/public/app_plugin/shared/edit_on_the_fly/lens_configuration_flyout.tsx

```diff
@@ -67,23 +74,30 @@ export function LensEditConfigurationFlyout({
   const activeVisualization = visualizationMap[attributes.visualizationType];
   const activeDatasource = datasourceMap[datasourceId];
   const { datasourceStates, visualization, isLoading } = useLensSelector((state) => state.lens);
+  const framePublicAPI = useLensSelector((state) => selectFramePublicAPI(state, datasourceMap));
   const suggestsLimitedColumns = activeDatasource?.suggestsLimitedColumns?.(datasourceState);
-  const activeData: Record<string, Datatable> = useMemo(() => {
-    return {};
-  }, []);
+
+  const layers = useMemo(
+    () => activeDatasource.getLayers(datasourceState),
+    [activeDatasource, datasourceState]
+  );
+
+  const dispatch = useLensDispatch();
   useEffect(() => {
     const s = output$?.subscribe(() => {
-      const layers = activeDatasource.getLayers(datasourceState);
-      const adaptersTables = lensAdapters?.tables?.tables as Record<string, Datatable>;
+      const activeData: Record<string, Datatable> = {};
+      const adaptersTables = previousAdapters.current?.tables?.tables as Record<string, Datatable>;
```

Comment:
@mbondyra are we sure that this is correct? I see this behavior. It seems as the chart and the editor are not synchronized or something like that.

![esql_bug](https://github.com/elastic/kibana/assets/17003240/a8e6bfa0-8770-4e35-953c-187edb69b348)

Reply:
of course!